src/methods/scprint/script.py

```diff
@@ -157,30 +171,41 @@
 # the Jaccard distances. This algorithm may not match all predicted labels so
 # we remove any that have been matched and repeat until all predicted labels are
 # matched to a dataset label.
-print("\n---- INFERRED MATCHES ----", flush=True)
-print(f"{'PREDICTED' : <40}{'LABEL' : <40}", flush=True)
-while not all(pred in matches for pred in predicted_levels):
-    # Get predicted labels that have not yet been matched
-    not_matched = [pred for pred in predicted_levels if pred not in matches.keys()]
-    not_matched_idx = [predicted_levels.index(pred) for pred in not_matched]
-
-    # Get assignments for currently unmatched predicted labels
-    assignments = linear_sum_assignment(jaccard[:, not_matched_idx])
-
-    # Store any new matches
-    for label, pred in zip(assignments[0], assignments[1]):
-        predicted_level = not_matched[pred]
-        label_level = label_levels[label]
-        matches[predicted_level] = label_level
-
-        if (len(predicted_level) > 39):
-            predicted_level = predicted_level[:36] + '...'
-
-        if (len(label_level) > 39):
-            label_level = label_level[:36] + '...'
-
-        print(f"{predicted_level: <40}{label_level: <40}", flush=True)
 
+print("\n---- INFERRED MATCHES ----", flush=True)
+print(f"{'PREDICTED': <40}{'LABEL': <40}", flush=True)
+
+
+if par["infer_matches"] == "direct":
+    # other version
+    for i, pred in enumerate(predicted_levels):
+        matches[pred] = label_levels[np.argmin(jaccard[:, i])]
+        print(f"{pred: <40}{matches[pred]: <40}", flush=True)
```

Comment:
Can you please explain how this works? I think you are mapping from the model to the labels from the dataset with the smallest Jaccard distance? If you have compared it to the linear sum assignment that would be interesting to know for other methods.

Reply:
I am sorry, I should have mentionned this. (I thought I had commented it out in the end) I tried it to see the difference. Overall it seems that linear sum assignment makes a bit more sense than direct assignment from jaccard distance. You get some label that don't match to their closest labels in linear sum. But you cover more labels. So this is a classic precision / recall problem. One would work better than the other depending on the metric

Reply:
It was a mistake to have my version by default and reverted to linear sum assignment

Reply:
Ok, cool. I was just curious if I understood how it worked. It's your method so I think it's fine to use whichever you prefer. There should maybe be an issue for how to handle assignment for models that have their own labels.

Reply:
I will keep it like this for now and might push an issue to add this secondary method as a another "method version".